telemetry/receivedenevlope.go

```diff
@@ -60,3 +61,45 @@ func (r *ReceivedEnvelope) updateProcessingError(db *sql.DB) error {
 
 	return nil
 }
+
+type SentEnvelope struct {
+	ID              int    `json:"id"`
+	MessageHash     string `json:"messageHash"`
+	SentAt          int64  `json:"sentAt"`
+	CreatedAt       int64  `json:"createdAt"`
+	PubsubTopic     string `json:"pubsubTopic"`
+	Topic           string `json:"topic"`
+	SenderKeyUID    string `json:"senderKeyUID"`
+	NodeName        string `json:"nodeName"`
+	ProcessingError string `json:"processingError"`
+	PublishMethod   string `json:"publishMethod"`
+	StatusVersion   string `json:"statusVersion"`
+}
+
+func (r *SentEnvelope) put(db *sql.DB) error {
+	r.CreatedAt = time.Now().Unix()
+	stmt, err := db.Prepare(`INSERT INTO sentEnvelopes (messageHash, sentAt, createdAt, pubsubTopic,
+							topic, senderKeyUID, nodeName, publishMethod, statusVersion)
+							VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)
+							ON CONFLICT ON CONSTRAINT sentEnvelopes_unique DO NOTHING
+							RETURNING id;`)
+	if err != nil {
+		return err
+	}
+
+	lastInsertId := int64(0)
+	res, err := stmt.Exec(r.MessageHash, r.SentAt, r.CreatedAt, r.PubsubTopic, r.Topic, r.SenderKeyUID, r.NodeName, r.PublishMethod, r.StatusVersion)
+	lastInsertId, _ = res.LastInsertId()
+
+	if err != nil {
```

Comment:
We should check for the error of stmt.Exec before retrieving the LastInsertId, 
Also, we should check for the error in res.LastInsertId instead of using `_`.

Reply:
I end up seeing this error:
```
Error saving sent envelope: LastInsertId is not supported by this driver Id:8
```

It looks like postgres does not support `LastInsertId`